datasets/switchboard.py

```diff
@@ -288,3 +298,35 @@ def _get_rec_to_segs_map(self):
                     rec_name = seg_info[0].split("-")[0]
                     rec_to_segs[rec_name].append(seg_info)
         return rec_to_segs
+
+
+class CreateSwitchboardLexiconTextFileJob(Job):
+    """
+    This job creates SWB preprocessed dictionary text file consistent with the training corpus given a raw dictionary
+    text file downloaded within the transcription directory using `DownloadSwitchboardTranscriptionAndDictJob` Job.
+    The resulted dictionary text file will be passed as argument to `LexiconFromTextFileJob` job in order to create
+    bliss xml lexicon.
+    """
+
+    def __init__(self, dict_dir):
+        """
+        :param tk.Path dict_dir: path containing the raw dictionary text file
+        """
+        self.dict_dir = dict_dir
+        self.out_dict = self.output_path("dict.txt")
+
+    def tasks(self):
+        yield Task("run", mini_task=True)
+
+    def run(self):
+        raw_dict_file = os.path.join(self.dict_dir.get_path(), "sw-ms98-dict.text")
+        with uopen(raw_dict_file) as read_f, uopen(self.out_dict, "w") as out_f:
+            for line in read_f.readlines()[1:]:
+                if line.startswith("#"):  # skip comment
+                    continue
+                parts = line.strip().split(" ", 1)
+                if len(parts) < 2:
+                    continue
+                token = parts[0].replace("&amp;", "&")  # e.g A&amp;E -> A&E
+                mapped_token = _map_token(token)  # preprocessing as corpus
+                out_f.write(mapped_token + " " + " ".join(parts[1:]) + "\n")
```

Comment:
why join here? you will have exactly 2 parts here

Reply:
ah yeah right